cran/cran.groovy

```diff
@@ -0,0 +1,609 @@
+    static final int INDEXING_SILENT_TIME = 5 * 1000l // 5 seconds
+    static final long REMOTE_INDEX_CACHE_TIME = 10 * 60 * 1000l // 10 minutes
+}
+
+executions {
+    /**
+     * Request repository CRAN indexing
+     * Parameters:
+     * repoKey - Key of CRAN repository
+     * path - Path to the artifacts parent folder to index
+     */
+    cranIndex(params: ['repoKey':[], 'path':[]]) { params ->
+        log.debug("cranIndex called with params: $params")
+
+        def repoKey = params['repoKey']?.get(0)
+        def path = params['path']?.get(0)
+
+        // Validate arguments
+        if (!repoKey) {
+            message = "Please inform the 'repoKey'"
+            status = 400
+            return
+        }
+        if (!path) {
+            message = "Please inform the 'path' to be indexed"
+            status = 400
+            return
+        }
+        if (!isCranRepo(repoKey)) {
+            message = "$repoKey is not a CRAN repository"
+            status = 400
+            return
+        }
+        def repoPath = RepoPathFactory.create(repoKey, path)
+        if (!repositories.exists(repoPath)) {
+            message = "CRAN repository $repoPath does not exist"
+            status = 400
+            return
+        }
+        def itemInfo = repositories.getItemInfo(repoPath)
+        if (!itemInfo.isFolder()) {
+            message = "$repoPath is not a folder"
+            status = 400
+            return
+        }
+
+        log.info("Indexing request received for CRAN repo $repoPath")
+        repositories.setProperty(repoPath, CRANConstants.INDEXING_STATUS_PROPERTY, 'scheduled')
+        executor.submit {
+            indexRepository(repoPath)
+        }
+        status = 200
+        message = "CRAN indexing execution requested for repo $repoPath"
+    }
+}
+
+download {
+    /**
+     * Expires remote repository cached index so the remote file will be used if available
+     */
+    beforeDownloadRequest { request, repoPath ->
+        log.trace ("beforeDownloadRequest: $repoPath")
+        if (repositories.getRemoteRepositories().contains(repoPath.getRepoKey()) &&
+                isRepositoryIndex(repoPath) &&
+                isFileExpired(repoPath)) {
+            log.info("Expiring remote repository index at " + repoPath)
+            expired = true
+        }
+    }
+}
+
+storage {
+    /**
+     * Handle creation of CRAN packages.
+     *
+     * At creation, extract package metadata and save it as artifact properties.
+     * If package is deployed at the repository's root folder, move it to the
+     * right location, according to package type. Otherwise, request indexing if needed.
+     */
+    afterCreate { item ->
+        log.trace("afterCreate: $item")
+        if (isCranRepo(item.repoPath.repoKey) && isCranPackage(item.repoPath)) {
+            log.info("Creation of CRAN package ${item.repoPath} detected")
+            storeCranPackageMetadata(item.repoPath)
+            if (item.repoPath.parent.root) {
```

Comment:
Apparently it works because the isRoot method has the following signature:
```java
boolean isRoot();
```
If the returning type was the Boolean wrapper it would probably throw an exception.